### PR TITLE
chore: update stylelint packages

### PR DESCRIPTION
**Pull request by dbrugger**
Closes #

Version upkeep for `stylelint-prettier`. Reason: I would like to use `stylelint-config-carbon` in apps. Some of these are already on Prettier 3, and getting stylelint-prettier to v4 is required for that. So this PR is a stepping stone.

#### Changelog

**Added**

- Instruction in README how to see effective changes

**Changed**

- stylelint-prettier to version 3
- stylelint-scss to latest
- stylelint-no-unsupported-browser-features to latest
- stylelint-config-standard-scss to latest
- stylelint-config-idiomatic-order

**Removed**

- Removed stylelint-order which is not used directly
- Removed stylelint-config-standard as it's used by stylelint-config-standard-scss (https://github.com/stylelint-scss/stylelint-config-standard-scss/blob/HEAD/index.js#L4)
- Removed deprecated rules per https://github.com/prettier/stylelint-prettier/blob/main/CHANGELOG.md#300-2023-02-22 and https://stylelint.io/user-guide/rules#deprecated which are covered by Prettier anyway
- plugins/prettier.js which is exactly the same like the recommended config from that plugin
- Removed `stylelint-config-prettier` as per its README:
> As of Stylelint v15 [all style-related rules have been deprecated](https://stylelint.io/migration-guide/to-15#deprecated-stylistic-rules). If you are using v15 or higher and are not making use of these deprecated rules, [this plugin is no longer necessary](https://stylelint.io/migration-guide/to-15#:~:text=Additionally%2C%20you%20may%20no%20longer%20need%20to%20extend%20Prettier%27s%20Stylelint%20config).

#### Testing / Reviewing

This is the effective change:

```
% npx json-diff --max-elisions 1 currentconfig.json newconfig.json
 {
   rules: {
-    at-rule-name-case: null
-    at-rule-name-space-after: null
-    at-rule-semicolon-newline-after: null
-    block-closing-brace-empty-line-before: null
-    block-closing-brace-newline-after: null
-    block-closing-brace-newline-before: null
-    block-closing-brace-space-before: null
-    block-opening-brace-newline-after: null
-    block-opening-brace-space-after: null
-    block-opening-brace-space-before: null
-    color-hex-case: null
-    declaration-bang-space-after: null
-    declaration-bang-space-before: null
-    declaration-block-semicolon-newline-after: null
-    declaration-block-semicolon-space-after: null
-    declaration-block-semicolon-space-before: null
-    declaration-block-trailing-semicolon: null
-    declaration-colon-newline-after: null
-    declaration-colon-space-after: null
-    declaration-colon-space-before: null
-    function-comma-newline-after: null
-    function-comma-space-after: null
-    function-comma-space-before: null
-    function-max-empty-lines: null
-    function-parentheses-newline-inside: null
-    function-parentheses-space-inside: null
-    function-whitespace-after: [
-      "always"
-    ]
-    indentation: null
-    max-empty-lines: null
-    max-line-length: null
-    media-feature-colon-space-after: null
-    media-feature-colon-space-before: null
-    media-feature-name-case: null
-    media-feature-parentheses-space-inside: null
-    media-feature-range-operator-space-after: [
-      "always"
-    ]
-    media-feature-range-operator-space-before: [
-      "always"
-    ]
-    media-query-list-comma-newline-after: null
-    media-query-list-comma-space-after: null
-    media-query-list-comma-space-before: null
-    no-empty-first-line: null
-    no-eol-whitespace: null
-    no-missing-end-of-source-newline: null
-    number-leading-zero: null
-    number-no-trailing-zeros: null
-    property-case: null
-    selector-attribute-brackets-space-inside: null
-    selector-attribute-operator-space-after: null
-    selector-attribute-operator-space-before: null
-    selector-combinator-space-after: null
-    selector-combinator-space-before: null
-    selector-descendant-combinator-no-non-space: null
-    selector-list-comma-newline-after: null
-    selector-list-comma-space-before: null
-    selector-pseudo-class-case: null
-    selector-pseudo-class-parentheses-space-inside: null
-    selector-pseudo-element-case: null
-    string-quotes: null
-    unit-case: null
-    value-list-comma-newline-after: null
-    value-list-comma-space-after: null
-    value-list-comma-space-before: null
-    value-list-max-empty-lines: null
-    no-extra-semicolons: null
-    function-comma-newline-before: null
-    value-list-comma-newline-before: null
-    declaration-block-semicolon-newline-before: null
-    block-closing-brace-space-after: null
-    block-opening-brace-newline-before: null
-    selector-list-comma-newline-before: null
-    selector-list-comma-space-after: null
-    media-query-list-comma-newline-before: null
-    at-rule-name-newline-after: null
-    at-rule-semicolon-space-before: null
-    linebreaks: null
-    unicode-bom: null
+    import-notation: [
+      "string"
+    ]
+    keyframe-selector-notation: [
+      "percentage-unless-within-keyword-only-block"
+    ]
+    media-feature-range-notation: [
+      "context"
+    ]
+    selector-not-notation: [
+      "complex"
+    ]
+    annotation-no-unknown: null
+    function-no-unknown: null
+    keyframe-block-no-duplicate-selectors: [
+      true
+    ]
+    selector-anb-no-unmatchable: [
+      true
+    ]
     keyframes-name-pattern: [
       ... (1 entries)
       {
-        message: "Expected keyframe name to be kebab-case"
       }
     ]
-    selector-attribute-quotes: null
+    selector-attribute-quotes: [
+      "always"
+    ]
     no-invalid-position-at-import-rule: [
       ... (1 entries)
       {
         ignoreAtRules: [
           ... (1 entries)
+          "forward"
         ]
       }
     ]
     order/properties-order: [
       [
         ... (38 entries)
+        "border-radius"
         ... (4 entries)
+        "border-top-radius"
         ... (4 entries)
+        "border-right-radius"
         ... (4 entries)
+        "border-bottom-radius"
         ... (4 entries)
+        "border-left-radius"
         ... (5 entries)
       ]
       ... (1 entries)
     ]
   }
   plugins: [
-    "/Users/dominik/carbon/node_modules/stylelint-config-idiomatic-order/node_modules/stylelint-order/index.js"
-    "/Users/dominik/carbon/node_modules/stylelint-scss/dist/index.js"
+    "/Users/dominik/carbon/node_modules/stylelint-scss/src/index.js"
+    "/Users/dominik/carbon/node_modules/stylelint-order/index.js"
+    "/Users/dominik/carbon/node_modules/stylelint-prettier/stylelint-prettier.js"
     ... (1 entries)
-    "/Users/dominik/carbon/node_modules/stylelint-prettier/stylelint-prettier.js"
   ]
 }

```

### CODE REVIEW

File: config/stylelint-config-carbon/package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "stylelint-config-standard-scss": "^3.0.0",
     "stylelint-no-unsupported-browser-features": "^5.0.2",
     "stylelint-order": "^5.0.0",
-    "stylelint-prettier": "^2.0.0",
+    "stylelint-prettier": "3",
```
Comment (tay1orjones) [inline config/stylelint-config-carbon/package.json]: ^3.0.0

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Thanks for this! I found one small thing to change.

> Removed deprecated rules per https://github.com/prettier/stylelint-prettier/blob/main/CHANGELOG.md#300-2023-02-22 and https://stylelint.io/user-guide/rules#deprecated which are covered by Prettier anyway

Covered by Prettier v2.x or v3.x? We won't want to have any lapse in rule coverage until we adopt Prettier v3. With that in mind, do we need to upgrade Prettier alongside these changes?

Also unless I'm mistaken, we will need to release this as a new major version for `stylelint-config-carbon`, correct?

**Comment by dbrugger** [comment]
Both prettier v2 and v3 cover the formatting of scss. That's why stylelint v15 deprecated these rules. There was a config `stylelint-config-prettier` which disabled these, which was useful before v15. Now this config is no longer required, as the rules are no longer used by default. That's why `stylelint-prettier` removed this config from its default.
I don't think a major version bump is required with this PR, as for the consumer nothing changes.
The only "interface" of this package is the peerDependency of stylelint v15. Once stylelint-prettier is updated to v4, which requires prettier v3 a major version bump is required imo, and then that should also include the prettier update.

**Comment by dbrugger** [comment]
@tay1orjones please have a look again. I now also added the other related packages, and took a systematic approach to see effective rule differences. In https://github.com/carbon-design-system/carbon/actions/runs/5936086642/job/16095800231?pr=14468 you see 27 new findings. We have to decide whether to ignore these new rules for now or to fix the offending scss.

Also a question. When I run `yarn lint:styles` locally it just hangs. The only way I can get it to run is if I add `--ignore-path .gitignore`, but then get a much higher number of findings than the pipeline. Can you run it locally? Is the pipeline doing anything special?

**Review by guidari — DISMISSED** [review]

**Comment by tay1orjones** [comment]
Thanks again for taking a jab at this! A few things have happened that I think might make this PR obsolete. 

1. Dependabot/renovate issued individual updates for multiple stylelint packages over the past few weeks. These PRs got these deps to their most current version, many of which are listed in the PR.
2. We've approached the prettier update from a new angle in https://github.com/carbon-design-system/carbon/pull/14744 and I think we'll land the update (and the related stylelint updates) there

For those reasons, I'm going to close this out. We can reopen if you think we've missed anything, or we could work together to get changes present here also present over in #14744. I think we've got them all, but let us know. Thank you again!

**Pull request closed without merging**